# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerInfo.h
# Commit: fb720435215f
# Full Hash: fb720435215fa5be3b2bd2101f3a54382954b70c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1544232
# File Overlap Count: 4
# Description:
#   Bug 1544232 - Limit lifetime extension of SWs by SWs to the sender's lifetime. r=edenchuang
#   
#   This patch introduces an explicit concept of lifetimes with mechanisms
#   in place so that actions taken by Clients (windows or non-ServiceWorker
#   orkers) will extend the lifetime of a ServiceWorker, but a ServiceWorker
# ==============================================================================

diff -r 25452f6623a0 -r fb720435215f dom/serviceworkers/ServiceWorkerInfo.h
--- a/dom/serviceworkers/ServiceWorkerInfo.h	Thu Oct 24 03:02:41 2024 +0000
+++ b/dom/serviceworkers/ServiceWorkerInfo.h	Thu Oct 24 03:02:42 2024 +0000
@@ -10,6 +10,7 @@
 #include "MainThreadUtils.h"
 #include "mozilla/dom/ServiceWorkerBinding.h"  // For ServiceWorkerState
 #include "mozilla/dom/ServiceWorkerDescriptor.h"
+#include "mozilla/dom/ServiceWorkerLifetimeExtension.h"
 #include "mozilla/dom/WorkerCommon.h"
 #include "mozilla/OriginAttributes.h"
 #include "mozilla/TimeStamp.h"
@@ -17,6 +18,7 @@
 
 namespace mozilla::dom {
 
+class ClientInfo;
 class PostMessageSource;
 class ServiceWorkerCloneData;
 class ServiceWorkerPrivate;
@@ -90,6 +92,12 @@
 
   const nsCString& Scope() const { return mDescriptor.Scope(); }
 
+  Maybe<ClientInfo> GetClientInfo();
+
+  // Pass-through of ServiceWorkerPrivate::GetLifetimeDeadline(); note that
+  // we have an XPCOM variation that returns a double for testing purposes.
+  TimeStamp LifetimeDeadline();
+
   bool SkipWaitingFlag() const {
     MOZ_ASSERT(NS_IsMainThread());
     return mSkipWaitingFlag;
@@ -108,7 +116,7 @@
   ServiceWorkerInfo(nsIPrincipal* aPrincipal, const nsACString& aScope,
                     uint64_t aRegistrationId, uint64_t aRegistrationVersion,
                     const nsACString& aScriptSpec, const nsAString& aCacheName,
-                    nsLoadFlags aLoadFlags);
+                    nsLoadFlags aImportsLoadFlags);
 
   ServiceWorkerState State() const { return mDescriptor.State(); }
 